src/rules/unit-no-unknown/index.js

```diff
@@ -0,0 +1,71 @@
+import { isString } from "lodash"
+import valueParser from "postcss-value-parser"
+
+import {
+  cssWordIsVariable,
+  declarationValueIndexOffset,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "unit-no-unknown"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (u) => `Unexpected unknown unit "${u}"`,
+})
+
+const knownUnits = new Set([
+  // Relative length units
+  "em", "ex", "ch", "rem", "%",
+  // Viewport-percentage lengths
+  "vh", "vw", "vmin", "vmax", "vm",
+  // Absolute length units
+  "px", "mm", "cm", "in", "pt", "pc",
+  // Time length units
+  "s", "ms",
+  // Angle
+  "deg", "grad", "turn", "rad",
+])
```

Comment:
In `number-zero-length-no-unit` there is set of length units. I think we should combine that and this set into a single reference in `utils`. That reference could be a map with categorized units or could be a function like `isKnownUnit()` that accepts an argument narrowing the units to check by category. Something like that. That way we would not have to maintain separate lists of units.